### PR TITLE
Handle namespaces individually when parsing for deprecated APIs

### PR DESCRIPTION
**Pull request by tkrishtop**

### CODE REVIEW

**Comment by tkrishtop** [comment]
check workload preflight-green

**Comment by tkrishtop** [comment]
Ready for review, [tested green](https://www.distributed-ci.io/jobs/21b70140-5c74-4bdb-ac9b-07be43d2278d/tests).
The idea of the change is to redesign the retrieval of the APIRequestCount object. 

APIRequestCount per namespace is about 100MB, and previously we only dealt with partners having about 2-3 workload namespaces. That allowed us to retrieve one large APIRequestCount for all workload namespaces simultaneously, creating one large object of about 300MB. [That stopped working on a partner having 22 custom namespaces - the task to retrieve APIRequestCount of about 2.2GB timed out](https://www.distributed-ci.io/jobs/dda0a2a4-4958-4368-b5bd-f5971a2cac06/jobStates?sort=date). 

This PR is to parse the namespaces for deprecated APIs one-by-one, to limit the size of APIRequestCount object.
Ex: [the job detects two workload namespaces]([apirequestcounts_ocp_compatibility_default.xml](https://www.distributed-ci.io/jobs/21b70140-5c74-4bdb-ac9b-07be43d2278d/tests/6a8b1c37-39e9-4a73-b808-0b203ce63f48) and generates separate compatibility testfile for each of them: [apirequestcounts_ocp_compatibility_default.xml](https://www.distributed-ci.io/jobs/21b70140-5c74-4bdb-ac9b-07be43d2278d/tests/6a8b1c37-39e9-4a73-b808-0b203ce63f48) and [apirequestcounts_ocp_compatibility_myns.xml](https://www.distributed-ci.io/jobs/21b70140-5c74-4bdb-ac9b-07be43d2278d/tests/c8421b8f-1e6b-4b71-b533-9c6de026dd40)

File: roles/deprecated_api/tasks/get_api_request_counts_per_namespace.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+- name: "Get API request counts for a workload ns {{ deprecated_ns }}"
```
Comment (manurodriguez) [inline roles/deprecated_api/tasks/get_api_request_counts_per_namespace.yml:2]: Nice findings about the APIRequestCount large object size!

In regards to that, in the successful job you reference it, I see the tasks `Get API request counts for a workload ns *` print a large output[1] and that is making the ansible.log heavier, would it be possible to add `no_log: true` here? Thanks

[1]
https://www.distributed-ci.io/jobs/21b70140-5c74-4bdb-ac9b-07be43d2278d/jobStates?sort=date&task=a490a05a-397c-4003-8e8c-a27ce35b2039
https://www.distributed-ci.io/jobs/21b70140-5c74-4bdb-ac9b-07be43d2278d/jobStates?sort=date&task=05b244ed-6628-4ca5-8ea1-bc97fbb66169

**Comment by tkrishtop** [comment]
check workload preflight-green

File: roles/deprecated_api/tasks/get_api_request_counts_per_namespace.yml (hunk 1 above)
Comment (tkrishtop) [reply roles/deprecated_api/tasks/get_api_request_counts_per_namespace.yml:2]: Hi @manurodriguez, thank you for the review and the nice idea! It looks logical, I don't know why I did not think of it. I pushed the change.

**Review by ramperher — APPROVED** [review]
LGTM

**Pull request merged**